core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchWriter.java

```diff
@@ -250,15 +229,24 @@ public synchronized void addMutation(TableId table, Mutation m)
 
     checkForFailures();
```

Comment:
Now that checkForFailures is not called in a synch block may throw ConcurrentModificationExceptions as its impl iterates over collections that other threads may modify.

Reply:
I added the synchronized modifier back to `addMutation`